diff --git a/node_modules/@helia/block-brokers/dist/src/trustless-gateway/broker.js b/node_modules/@helia/block-brokers/dist/src/trustless-gateway/broker.js
index 223f634..34ff22d 100644
--- a/node_modules/@helia/block-brokers/dist/src/trustless-gateway/broker.js
+++ b/node_modules/@helia/block-brokers/dist/src/trustless-gateway/broker.js
@@ -58,10 +58,14 @@ export class TrustlessGatewayBlockBroker {
         }
     }
     createSession(options = {}) {
-        return createTrustlessGatewaySession({
-            logger: this.logger,
-            routing: this.routing
-        }, options);
+      return createTrustlessGatewaySession({
+        logger: this.logger,
+        routing: this.routing
+      }, {
+        ...options,
+        allowLocal: this.allowLocal,
+        allowInsecure: this.allowInsecure
+      })
     }
 }
 //# sourceMappingURL=broker.js.map
\ No newline at end of file
